Services/TMS/TrainingSearch/classes/class.ilTrainingSearchTableGUI.php

```diff
@@ -85,6 +86,25 @@ function ($row, BookableCourse $record, $ui_factory, $environment) { //mapping-c
 		//apply data to table and render
 		return $renderer->render($ptable->withData($data));
 	}
+
+	/**
+	 * Get all sorting and filter items for the table
+	 *
+	 * @param 	$f
+	 *
+	 * @return Sortation[]
+	 */
+	protected function getSortationObjects($f) {
```

Comment:
maybe better set $this->ui_factory = $DIC->ui()->factory() in the constructor;
also in render().